handlers/grafana_handlers.go

```diff
@@ -28,8 +29,13 @@ func (h *Handler) GrafanaConfigHandler(w http.ResponseWriter, req *http.Request,
 	if req.Method == http.MethodPost {
 		grafanaURL := req.FormValue("grafanaURL")
 		grafanaAPIKey := req.FormValue("grafanaAPIKey")
-		if strings.Contains(grafanaURL, "/graph") {
-			grafanaURL = strings.TrimSuffix(grafanaURL, "/graph")
+
+		u, err := url.Parse(grafanaURL)
+		if err != nil {
+			return
```

Comment:
@subhamkrai thanks. What will happen if the parser errors out? We may want to send a signal to the UI to let them know that it is an invalid URL.

Reply:
Yes can do that too